### PR TITLE
[Server] Fix hard coded WFS version in describe feature type URL

### PR DESCRIPTION
**Pull request by rldhont**
## Description

The WFS version has been hardcoded in the describe feature type URL provided in the GetCapabilities.

Issue mentionned in https://github.com/qgis/QGIS/issues/37555#issuecomment-748198725

### CODE REVIEW

**Comment by symbioquine** [comment]
@rldhont minor correction on the description here. The DescribeFeatureType URI is provided as part of the Transaction request, not GetCapabilities.

File: src/providers/wfs/qgswfsprovider.cpp [hunk 1]
```diff
@@ -1657,7 +1657,15 @@ QDomElement QgsWFSProvider::createTransactionElement( QDomDocument &doc ) const
     describeFeatureTypeURL.setQuery( query );
   }
   QUrlQuery query( describeFeatureTypeURL );
-  query.addQueryItem( QStringLiteral( "VERSION" ), QStringLiteral( "1.0.0" ) );
+  // only 1.1.0 and 1.0.0 are supported
+  if ( WfsVersion == QStringLiteral( "1.1.0" ) )
+  {
+    query.addQueryItem( QStringLiteral( "VERSION" ), WfsVersion );
```
Comment (elpaso) [inline src/providers/wfs/qgswfsprovider.cpp:1663]: Is the `if/else` necessary? Why cannot we just always do `query.addQueryItem( QStringLiteral( "VERSION" ), WfsVersion );` ?

Comment (symbioquine) [reply src/providers/wfs/qgswfsprovider.cpp:1663]: My understanding is that it matches [this logic](https://github.com/qgis/QGIS/blob/13e4969817e990629748a3342938df0c772e14c4/src/providers/wfs/qgswfsprovider.cpp#L1639-L1646) above. Though one could make an argument for storing the "effective version" in a variable as part of that first check and using it thereafter...

Comment (elpaso) [reply src/providers/wfs/qgswfsprovider.cpp:1663]: Ah, I see now it's from the line 1639 above. Maybe you could set a version string there and avoid code duplication?

Comment (rldhont) [reply src/providers/wfs/qgswfsprovider.cpp]: Code updated

**Comment by rldhont** [comment]
Travis failed on PyQgsWFSProvider, but I don't think I have made changes that impact it.

**Comment by Gustry** [comment]
> Travis failed on PyQgsWFSProvider, but I don't think I have made changes that impact it.

Strange indeed, but it looks quite close

**Pull request closed without merging**